src/dark_mode_toggle.tsx

```diff
@@ -126,7 +126,22 @@ export function DarkModeToggle({ api }: DarkModeProps) {
 	const setMode = useCallback(
 		(mode: Mode) => {
 			const currentStore = store()
-			api.setOptions({ theme: currentStore[mode] })
+			// Get current addon configuration to preserve existing theme customizations
+			const currentConfig = addons.getConfig()
+			const currentTheme = currentConfig?.theme || {}
+			
+			// Merge new theme with existing theme to preserve user customizations
+			// This prevents CSS specificity conflicts by maintaining brand properties
+			const newTheme = {
+				...currentStore[mode],
+				// Preserve brand-related properties from existing theme
+				...(currentTheme.brandImage && { brandImage: currentTheme.brandImage }),
```

Comment:
These logic can probably be simplified. But no big deal.

The concern I have is as detailed in the comment in the OP, which this PR fixes for these 4 fields. There may be more to fix now on in the future.